agents/model_capabilities.py

```
@@ -0,0 +1,179 @@
+        return None
+    return _ollama_show(model_name, base.rstrip("/"))
+
+
+@lru_cache(maxsize=64)
+def _ollama_show(model_name: str, base_url: str) -> tuple[int, int] | None:
+    try:
+        req = urllib.request.Request(
+            f"{base_url}/api/show",
+            data=json.dumps({"model": model_name}).encode(),
+            headers={"Content-Type": "application/json"},
+        )
```

Ruff (0.15.10)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)